packages/tpetra/core/ext/TpetraExt_MatrixMatrix_def.hpp

```diff
@@ -1563,8 +1571,11 @@ void mult_AT_B_newmatrix(
     if(!params.is_null()) {
         Teuchos::ParameterList& params_sublist = params->sublist("matrixmatrix: kernel params",false);
         Teuchos::ParameterList& labelList_subList = labelList.sublist("matrixmatrix: kernel params",false);
-        int foundcount = params_sublist.get("MM_TAFC_OptimizationCoreCount",3000);
-        labelList_subList.set("MM_TAFC_OptimizationCoreCount",foundcount,"Core Count above which the optimized neighbor discovery is used");
+        int mm_optimization_core_count = 3000;
+        mm_optimization_core_count = params_sublist.get("MM_TAFC_OptimizationCoreCount",mm_optimization_core_count);
+        int foo = params->get("MM_TAFC_OptimizationCoreCount",mm_optimization_core_count);
+        if(foo<mm_optimization_core_count) mm_optimization_core_count=foo;
+        labelList_subList.set("MM_TAFC_OptimizationCoreCount",mm_optimization_core_count,"Core Count above which the optimized neighbor discovery is used");
 
         labelList_subList.set("isMatrixMatrix_TransferAndFillComplete",true,
                       "This parameter should be set to true only for MatrixMatrix operations: the optimization in Epetra that was ported to Tpetra does _not_ take into account the possibility that for any given source PID, a particular GID may not exist on the target PID: i.e. a transfer operation. A fix for this general case is in development.");
```

Comment:
Instead of this "is in development" promise, it might be better to file a GitHub issue and point to the issue.  That will ensure proper tracking and also help gather feedback on whether users want this feature.